File: languages/js/src/entity/identity_v2/integer_attribute.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities_v2";
import { Attribute } from "./attribute";

/**
 * Represents an attribute with an integer value.
 */
export class IntegerAttribute extends Attribute<number> {
  public toProto(): identityEntitiesProto.IntegerAttributeV2 {
    return identityEntitiesProto.IntegerAttributeV2.fromPartial({
      id: this.id,
      value: this.value
    });
  }

  static fromProto(
    r: identityEntitiesProto.IntegerAttributeV2
  ): IntegerAttribute {
    return new IntegerAttribute(r.id, r.value);
  }
}

File: languages/java/src/main/java/com/bloock/sdk/entity/identity/IntegerAttribute.java
package com.bloock.sdk.entity.identity;

import com.bloock.sdk.bridge.proto.BloockIdentityEntities;

/**
 * Represents an attribute with an integer value.
 */
public class IntegerAttribute extends Attribute<Long> {
  /**
   * Creates a new IntegerAttribute instance with the provided key and value.
   * 
   * @param id
   * @param value
   */
  public IntegerAttribute(String id, Long value) {
    super(id, value);
  }

  public static IntegerAttribute fromProto(BloockIdentityEntities.IntegerAttribute res) {
    return new IntegerAttribute(res.getId(), res.getValue());
  }

  public BloockIdentityEntities.IntegerAttribute toProto() {
    return BloockIdentityEntities.IntegerAttribute.newBuilder()
        .setId(this.id)
        .setValue(this.value)
        .build();
  }
}
